### ISSUE TITLE
Improve ordering of matches for nodes endpoint

### ISSUE THREAD

**Issue by dhimmel** [greenelab/connectivity-search-backend#49]
Currently, searching `ep` will return `nicotine dependence` before `epilepsy`.

We are also thinking of including number of metapaths in database functionality.

**Comment by dhimmel** [comment]
##### Notes to self about django-rest-framework

https://www.django-rest-framework.org/api-guide/filtering/#searchfilter

```python
    search_fields = ('identifier', '^metanode__identifier', 'name', '=id')
```

**Comment by vincerubinetti** [comment]
I believe there's two separate issues here. @dhimmel I created a separate issue for the ordering of search results #52 . Could we make this issue more specifically about the number of metapaths?

Here's how I imagine the functionality:

The user already has a source node selected. They focus the target node search box (with nothing typed in yet). This is the scenario where I populate the autosuggest list with the highest "scoring" target nodes for the given source node. When the user starts to type, then the list is populated with the most "relevant" nodes to the string, based on some criteria as discussed in #52 (that criteria perhaps could be a balance between the relevance and the node "score").

If both the source and target search boxes are empty, when the user focuses one without having typed anything in yet, perhaps we could populate the autosuggest list with the highest "scoring" nodes in the hetmech database overall, perhaps based on metapaths or connectivity or something like that.

**Comment by vincerubinetti** [comment]
> Currently, searching `ep` will return `nicotine dependence` before `epilepsy`.

This is captured by issue #52.

> We are also thinking of including number of metapaths in database functionality.

This is closed by PR #51

**Comment by dongbohu** [comment]
Fixed by #59.
